src/test/java/com/aws/greengrass/clientdevices/auth/iot/usecases/VerifyThingAttachedToCertificateTest.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.iot.usecases;
+
+import com.aws.greengrass.clientdevices.auth.exception.CloudServiceInteractionException;
+import com.aws.greengrass.clientdevices.auth.infra.NetworkState;
+import com.aws.greengrass.clientdevices.auth.iot.IotAuthClient;
+import com.aws.greengrass.clientdevices.auth.iot.Thing;
+import com.aws.greengrass.clientdevices.auth.iot.dto.VerifyThingAttachedToCertificateDTO;
+import com.aws.greengrass.clientdevices.auth.iot.infra.ThingRegistry;
+import com.aws.greengrass.testcommons.testutilities.GGExtension;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith({MockitoExtension.class, GGExtension.class})
+class VerifyThingAttachedToCertificateTest {
+    @Mock
+    private IotAuthClient mockIotAuthClient;
+    @Mock
+    private NetworkState mockNetworkState;
+    @Mock
+    private ThingRegistry mockThingRegistry;
+    private VerifyThingAttachedToCertificate verifyThingAttachedToCertificate;
+
+    @BeforeEach
+    void beforeEach() {
+        verifyThingAttachedToCertificate = new VerifyThingAttachedToCertificate(mockIotAuthClient,
+                mockThingRegistry, mockNetworkState);
+    }
+
+    @Test
+    void GIVEN_validDtoAndNetworkUp_WHEN_verifyThingAttachedToCertificate_THEN_returnCloudResult() {
+        Boolean expectedCloudResult = true;
+        Thing mockThing = Thing.of("mock-thing");
+        String mockCertId = "cert-id";
+        VerifyThingAttachedToCertificateDTO dto =
+                new VerifyThingAttachedToCertificateDTO(mockThing.getThingName(), mockCertId);
+
+        when(mockNetworkState.getConnectionStateFromMqtt()).thenReturn(NetworkState.ConnectionState.NETWORK_UP);
+        when(mockIotAuthClient.isThingAttachedToCertificate(any(), anyString())).thenReturn(expectedCloudResult);
```

Comment:
Why not use the IotAuthFake?. I know this is a unit test but the less we mock as a general rule the better given our tests become less coupled to implementation while giving us confidence that things are working.